build/gen_stub.php

```diff
@@ -1480,23 +1483,380 @@ private function getFlagsAsArginfoString(): string
+        } else {
+            $returnDescriptionPara->appendChild(new DOMText("Description"));
+        }
+        $returnDescriptionPara->appendChild(new DOMText("\n  "));
+        $returnRefSec->appendChild($returnDescriptionPara);
+        $returnRefSec->appendChild(new DOMText("\n "));
+        return $returnRefSec;
+    }
+
+    /**
+     * @param array<DOMNode> $headers [count($headers) === $columns]
+     * @param array<array<DOMNode>> $rows [count($rows[$i]) === $columns]
+     */
+    private function generateDocbookInformalTable(
+        DOMDocument $doc,
+        int $indent,
+        int $columns,
+        array $headers,
+        array $rows
+    ): DOMElement {
+        $strIndent = str_repeat(' ', $indent);
+
+        $headerRow = $doc->createElement('row');
+        foreach ($headers as $header) {
+            $headerEntry = $doc->createElement('entry');
+            $headerEntry->appendChild($header);
+
+            $headerRow->appendChild(new DOMText("\n$strIndent    "));
+            $headerRow->appendChild($headerEntry);
+        }
+        $headerRow->appendChild(new DOMText("\n$strIndent   "));
+
+        $thead = $doc->createElement('thead');
+        $thead->appendChild(new DOMText("\n$strIndent   "));
+        $thead->appendChild($headerRow);
+        $thead->appendChild(new DOMText("\n$strIndent  "));
+
+        $tbody = $doc->createElement('tbody');
+        foreach ($rows as $row) {
+            $bodyRow = $doc->createElement('row');
+            foreach ($row as $cell) {
+                $entry = $doc->createElement('entry');
+                $entry->appendChild($cell);
+
+                $bodyRow->appendChild(new DOMText("\n$strIndent    "));
+                $bodyRow->appendChild($entry);
+            }
+            $bodyRow->appendChild(new DOMText("\n$strIndent   "));
+
+            $tbody->appendChild(new DOMText("\n$strIndent   "));
+            $tbody->appendChild($bodyRow);
+            $tbody->appendChild(new DOMText("\n$strIndent  "));
+        }
+
+        $tgroup = $doc->createElement('tgroup');
+        $tgroup->setAttribute('cols', (string) $columns);
+        $tgroup->appendChild(new DOMText("\n$strIndent  "));
+        $tgroup->appendChild($thead);
+        $tgroup->appendChild(new DOMText("\n$strIndent  "));
+        $tgroup->appendChild($tbody);
+        $tgroup->appendChild(new DOMText("\n$strIndent "));
+
+        $table = $doc->createElement('informaltable');
+        $table->appendChild(new DOMText("\n$strIndent "));
+        $table->appendChild($tgroup);
+        $table->appendChild(new DOMText("\n$strIndent"));
+
+        return $table;
+    }
+
+    private function getChangelogSection(DOMDocument $doc): DOMElement {
+        $refSec = $this->generateRefSect1($doc, 'changelog');
+        $headers = [
+            $doc->createEntityReference('Version'),
+            $doc->createEntityReference('Description'),
+        ];
+        $rows = [[
+            new DOMText('8.X.0'),
```

Comment:
overengineering of the day: it would be nice to be able to retrieve the current PHP version in development via a web service :D

Reply:
Yeah I wasn't using the sapi/cli binary, but my locally installed version